src/scripts/migrate.py

```diff
@@ -96,25 +96,31 @@ def slurp(
     unmapped_terms: List[Term] = [x for x in owned_terms if x.curie not in sssom_object_ids]
     slurp_candidates = [x for x in unmapped_terms if x.curie not in excluded_terms]  # remove exclusions
 
-    # Determine slurpable terms
-    # Migratable term: A term owned by the given ontology but unmapped in Mondo, which has either no parents, or if it
-    # has parents, all of its parents have already been mapped in Mondo.
+    # Determine slurpable / migratable terms
+    # To be migratable, the term (i) must not already be mapped, (ii) must not be excluded (e.g. not in
+    # `reports/%_term_exclusions.txt`), and (iii) must not be deprecated / obsolete. Then, (iv) there are conditions
+    # concerning its parents. It must either (a) have no parents, or (b) have no valid parents in Mondo (i.e. all of its
+    # parent terms are marked obsolete in Mondo), or (c) all its parents must be mapped, and at least 1 of those
+    # parent's mappings must be one of `skos:exactMatch` or `skos:NarrowMatch`.
     terms_to_slurp: List[Dict[str, str]] = []
     for t in slurp_candidates:
-        # If all T.parents mapped, and 1+ is an exact or narrow match and non obsolete, designate T for slurping
-        # (i.e. only slurp if parents are already slurped)
+        obsolete_mondo_parent_ids = []
         qualified_parent_mondo_ids = []
+        all_parents_mapped = True
         no_parents: bool = not t.direct_owned_parent_curies
         for parent_curie in t.direct_owned_parent_curies:
-            # Conversely, if any of T.parents is unmapped, T is not to be slurped
+            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:  # owned check redudnant?
+                all_parents_mapped = False
             if parent_curie in sssom_object_ids:
                 parent: Dict = sssom_df[sssom_df['object_id'] == parent_curie].to_dict('records')[0]
                 parent_mondo_id, parent_mondo_label = parent['subject_id'], parent['subject_label']
-                if not parent_mondo_label.startswith('obsolete'):
+                if parent_mondo_label.startswith('obsolete'):
+                    obsolete_mondo_parent_ids.append(parent_mondo_id)
+                elif parent['predicate_id'] in ['skos:exactMatch', 'skos:narrowMatch']:
                     qualified_parent_mondo_ids.append(parent_mondo_id)
-            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:
-                break
-        if qualified_parent_mondo_ids or no_parents:
+
+        no_valid_mondo_parents: bool = len(t.direct_owned_parent_curies) == len(obsolete_mondo_parent_ids)
+        if (all_parents_mapped and qualified_parent_mondo_ids) or no_parents or no_valid_mondo_parents:
```

Comment:
@matentzn Here's the if statement where it checks to make sure the conditions for determining if the term should be added to slurp. Most importantly, can you carefully look over what I wrote in the docs? And then if you want to double check my logic here to make sure it matches, feel free. But I think it is correct. 

Some things in particular that I want you to know:
1. Bugfix: Obsolete parents: I believe I fixed correctly as instructed. (i) if just some of the parents are obsolete in Mondo, a term can still be slurped, (ii) if _all_ of the parents are obsolete in Mondo, it will it will go in the slurp.
2. Bugfix: All parents must be mapped: There was a bug here, sorry. If a parent was not mapped, the loop would previously break, but this would not always have the intended effect. So some things may have slipped into slurp previously that shouldn't have been there.
3. Bugfix: `skos:exactMatch` or `skos:narrowMatch`: I wrote in the comments that at least 1 of the parents must be mapped with one of these mapping predicates, but I never added the check. It considered any mapping to be OK. In practice, I don't know if this would have happened very often, but the logic is there now to catch it.